crates/api/src/comment.rs

```diff
@@ -64,6 +64,19 @@ impl Perform for CreateComment {
       return Err(APIError::err("locked").into());
     }
 
+    // If there's a parent_id, check to make sure that comment is in that post
+    if let Some(parent_id) = data.parent_id {
+      // Make sure the parent comment exists
+      let parent =
+        match blocking(context.pool(), move |conn| Comment::read(&conn, parent_id)).await? {
+          Ok(comment) => comment,
+          Err(_e) => return Err(APIError::err("couldnt_create_comment").into()),
+        };
+      if parent.post_id != post_id {
+        return Err(APIError::err("couldnt_create_comment").into());
+      }
+    }
+
```

Comment:
At some point we need to return better error messages, its gonna be pretty frustrating for client devs figuring out which one of 4 different `couldnt_create_comment` errors they triggered.

Reply:
Good point. I'll add another error message for this one and open up a PR.